Add dig replacement

MadderHat opened the pull request:
Dig introducers to get available connections and automatically replace port 8444 connections

MadderHat committed:
Add dig replacement

Dig introducers to get available connections and automatically replace port 8444 connections

RightSexyOrc commented:
Hi, just took a quick look at it, first 2 things to pop out at me are:

1) pls include our 2 newest introducers, or i can add them after

  dns_servers: &id003
  - dns-introducer.aba.ooo
  - dns-introducer.abacoin.io
  - dns-introducer.abacoin.net
  - dns-introducer.aba-project.io
  

2) introducer var is not actually used in this loop:
"        # Run dig on introducers
        for introducer in INTRODUCERS:
            # Use +short to only get the IP address so we don't have to mess with the header/footer
            result = subprocess.run(['dig', 'dns-introducer.aba.ooo', '+short'], 
                                    capture_output=True, text=True, check=True)
            lines = result.stdout.split('\n')
            for line in lines:
                # Check if peer exists as aba peer
                if not check_existing_aba_peer(line, existing_connections):
                    if line != '' and not line in connections:
                    	connections.append(line)"

(something that may help: if using an llm ide, ask it to do a code review of your code, it may also help spot something else; i wonder if it spots this one too ), i can also potentially fix this too; will look at it in more detail later also

MadderHat committed:
Additional introducers and correctly use them

Added additional introducers and update line 43 to use the introducer var to correctly use all of the introducers

MadderHat commented:
I've added the introducers and fixed the introducer loop

MadderHat committed:
LLM assisted update

Update to previous commit as analyzed by Grok3.0
https://x.com/i/grok/share/FC696N10Aqgp4IaHIpBaC49w2

Errors Fixed: 
 * indexing in Peer Replacement
 * Incorrect Replacement Logic (related to above)
 * Potential Duplicate Peers
 * Fragile Peer List parsing
Vulnerabilities Fixed:
 * Subprocess Command Safety
 * Command and venv verification
Improvements:
 * Enhanced Dry-Run mode
 * Configurable Ports
 * Successful Subprocess Verification
Not implemented:
 * Exposure to Sensitive Information - All information is public
 * User Confirmation - This would be tedious
 * Logging - Overcomplication for this application
 * Summary Report - Grok missed this already existed

MadderHat commented:
Oh, dangit. I forgot to add "Reordered introducer list to put newest first" to the commit "Improvements" description before I hit commit!
